### PR TITLE
Windows: support embedded utility VM images

### PR DESCRIPTION
**Pull request by jstarks**
For Hyper-V containers, a utility VM image is necessary to start the VM that hosts the container. In TP5, this is distributed with the base layer.

With this change, when launching a container that uses a Docker-managed base layer, the utility VM is provided if it is present. Without this, Hyper-V containers cannot work with Docker-managed base layers.

For Windows-managed base layers (currently the expected user workflow), nothing changes -- the utility VM image is automatically found by Windows.

The details of this will probably change post-TP5.

cc/@jhowardmsft

### CODE REVIEW

File: daemon/oci_windows.go [hunk 1]
```diff
@@ -109,6 +97,34 @@ func (daemon *Daemon) createSpec(c *container.Container) (*libcontainerd.Spec, e
...
+		// Container is set to use the default, so take the default from the daemon configuration
+		hv = daemon.defaultIsolation.IsHyperV()
+	} else {
+		// Container is requesting an isolation mode. Honour it.
+		hv = c.HostConfig.Isolation.IsHyperV()
+	}
+	if hv {
+		hvr := &windowsoci.HvRuntime{}
+		if img.RootFS != nil && img.RootFS.Type == image.TypeLayers {
+			// For TP5, the utility VM is part of the base layer.
```
Comment (thaJeztah) [inline daemon/oci_windows.go]: Do we need a `TODO Windows...` here if this is to be changed after TP5?

Comment (jstarks) [inline daemon/oci_windows.go]: Good idea, I've added a TODO.

**Comment by jhowardmsft** [comment]
LGTM

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**